source/val/validate_extensions.cpp

```diff
@@ -149,10 +149,11 @@ spv_result_t ValidateExtInst(ValidationState_t& _, const Instruction* inst) {
              ++operand_index) {
           const uint32_t operand_type = _.GetOperandTypeId(inst, operand_index);
           if (result_type != operand_type) {
-            return _.diag(SPV_ERROR_INVALID_DATA, inst)
-                   << ext_inst_name() << ": "
-                   << "expected types of all operands to be equal to Result "
-                      "Type";
+            // Workaround a DXC generation error.
+            _.diag(SPV_ERROR_INVALID_DATA, inst)
+                << ext_inst_name() << ": "
+                << "expected types of all operands to be equal to Result "
+                   "Type";
```

Comment:
This does not look right.  This seems like DXC is generating invalid code, and you are letting it through.  This could be the cause of one of the errors in constant folding.  Is there a DXC bug that is open for this?  We could try to get that fixed if there is.